src/analysis/normed_space/ordered.lean

```diff
@@ -48,3 +67,146 @@ instance : normed_linear_ordered_field ℚ :=
 noncomputable
 instance : normed_linear_ordered_field ℝ :=
 ⟨dist_eq_norm, normed_field.norm_mul⟩
+
+/-!
+### Normed lattice orderd groups
+
+Motivated by the theory of Banach Lattices, this section introduces normed lattice ordered groups.
+-/
+
+local notation `|`a`|` := abs a
+
+/--
+Let `α` be a normed commutative group equipped with a partial order covariant with addition, with
+respect which `α` forms a lattice. Suppose that `α` is *solid*, that is to say, for `a` and `b` in
+`α`, with absolute values `|a|` and `|b|` respectively, `|a| ≤ |b|` implies `∥a∥ ≤ ∥b∥`. Then `α` is
+said to be a normed lattice ordered group.
+-/
+class normed_lattice_add_comm_group (α : Type*)
+  extends normed_group α, lattice α :=
+(add_le_add_left : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b)
+(solid : ∀ a b : α, |a| ≤ |b| → ∥a∥ ≤ ∥b∥)
+
+/--
+A normed lattice ordered group is a lattice ordered group
+-/
+instance normed_lattice_add_comm_group_to_covariant_class {α : Type*}
+  [h : normed_lattice_add_comm_group α] : covariant_class α α (+) (≤) :=
+{elim := λ a b c bc,  normed_lattice_add_comm_group.add_le_add_left _ _ bc a}
+
+/--
+Let `α` be a normed group with a partial order. Then the order dual is also a normed group.
+-/
+instance {α : Type*} : Π [normed_group α], normed_group (order_dual α) := id
+
+/--
+Let `α` be a normed lattice ordered group and let `a`and `b` be elements of `α`. Then `a⊓-a ≥ b⊓-b`
+implies `∥a∥ ≤ ∥b∥`.
+-/
+lemma dual_solid {α : Type*} [h: normed_lattice_add_comm_group α] : ∀ a b : α, b⊓-b ≤ a⊓-a →
+  ∥a∥ ≤ ∥b∥ :=
+begin
+  intros a b h₁,
+  apply h.solid,
```

Comment:
Rather than using `h.solid`, make a separate definition, something like

```
lemma solid [normed_lattice_add_comm_group α] {a b : α} (h : |a| ≤ |b|) : ∥a∥ ≤ ∥b∥ :=
normed_lattice_add_comm_group.solid
```

Reply:
Okay, done, but why is this better?

Reply:
It's not intrinsically much better, but it is the way we set up the API for all typeclasses, and it's good to make things unsuprising!